core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -74,14 +165,37 @@ public static void processPartition(
             final long newPartitionSize,
             final long oldPartitionSize,
             long partitionUpdateSinkAddr,
-            long dedupColSinkAddr
+            long dedupColSinkAddr,
+            boolean isParquet
     ) {
         // is out of order data hitting the last partition?
         // if so we do not need to re-open files and write to existing file descriptors
         final long o3TimestampLo = getTimestampIndexValue(sortedTimestampsAddr, srcOooLo);
         final long o3TimestampHi = getTimestampIndexValue(sortedTimestampsAddr, srcOooHi);
         final RecordMetadata metadata = tableWriter.getMetadata();
         final int timestampIndex = metadata.getTimestampIndex();
+        if (isParquet) {
+            processParquetPartition(
+                    pathToTable,
+                    partitionBy,
+                    oooColumns,
+                    srcOooLo,
+                    srcOooHi,
+                    partitionTimestamp,
+                    sortedTimestampsAddr,
+                    tableWriter
+            );
+
+            Unsafe.getUnsafe().putLong(partitionUpdateSinkAddr, partitionTimestamp);
+            Unsafe.getUnsafe().putLong(partitionUpdateSinkAddr + Long.BYTES, o3TimestampMin);
+            Unsafe.getUnsafe().putLong(partitionUpdateSinkAddr + 2 * Long.BYTES, newPartitionSize);
+            Unsafe.getUnsafe().putLong(partitionUpdateSinkAddr + 3 * Long.BYTES, oldPartitionSize);
+            Unsafe.getUnsafe().putLong(partitionUpdateSinkAddr + 4 * Long.BYTES, 1); // partitionMutates
+            Unsafe.getUnsafe().putLong(partitionUpdateSinkAddr + 5 * Long.BYTES, 0); // o3SplitPartitionSize
+            tableWriter.o3ClockDownPartitionUpdateCount();
+            return;
```

Comment:
We need to pass the resulting parquet file to be saved to `_txn` file so that TableReaders can concurrently read the file while we append to it. Should be passed somewhere in `partitionUpdateSinkAddr` and processed in TableWriter.